jobs/pulse3d/pulse3d-worker/src/main.py

```diff
@@ -200,7 +203,24 @@ async def process(con, item):
                         well_force = recording_df[well].dropna().tolist()
 
                         interpolated_well_data = np.row_stack([time[: len(well_force)], well_force])
-                        peaks, valleys = peak_detector(interpolated_well_data, **peak_detector_args)
+
+                        try:
+                            # Tanner (2/17/23): just need to get a quick fix out and not sure of all edge cases for this, so wrapping in try/except to be safe
+                            # apply window before running peak detection
+                            start_time_us = analysis_params.get("start_time", 0) * MICRO_TO_BASE_CONVERSION
+                            end_time_us = analysis_params.get("end_time", np.inf) * MICRO_TO_BASE_CONVERSION
+                            indices_after_start = interpolated_well_data[0] >= start_time_us
+                            indices_before_end = interpolated_well_data[0] <= end_time_us
+
+                            indices_to_keep = indices_after_start & indices_before_end
+                            windowed_well_data = interpolated_well_data[:, indices_to_keep]
+
+                            peaks, valleys = peak_detector(windowed_well_data, **peak_detector_args)
+                            idx_shift = np.argmax(indices_after_start)
+                            peaks += idx_shift
+                            valleys += idx_shift
```

Comment:
SO this is shifting the peaks and valley indices back to match the non windowed data?

Reply:
Yes, not ideal but I just want to get something out the works. I'm working on a better solution now that requires a new pulse3d version

Reply:
Yeah just making sure I knew what was happening